lib/ansible/plugins/become/sudo.py

```diff
@@ -94,7 +94,13 @@ def build_become_command(self, cmd, shell):
         if self.get_option('become_pass'):
             self.prompt = '[sudo via ansible, key=%s] password:' % self._id
             if flags:  # this could be simplified, but kept as is for now for backwards string matching
-                flags = flags.replace('-n', '')
+                for flag in list(flags.split(' ')):
```

Comment:
This could cause issues if there is anything in the args that might be a path with a space in it.  I'd really suggest using `shlex.split` instead, which will respect any quoting.  I think spliting into a list via `shlex.split`, and then rejoining will ultimately be the most safe.